akka-actor/src/main/scala/akka/actor/ActorRef.scala

```diff
@@ -599,6 +599,22 @@ private[akka] class VirtualPathContainer(
 
   private val children = new ConcurrentHashMap[String, InternalActorRef]
 
+  override def !(message: Any)(implicit sender: ActorRef = Actor.noSender): Unit = message match {
+    case sel @ ActorSelectionMessage(msg, elements, _) ⇒ {
+      elements.head match {
+        case SelectChildName(name) ⇒
+          val child = getChild(name)
+          if (elements.tail.isEmpty) {
+            child ! msg
+          } else {
+            child ! sel.copy(elements = elements.tail)
+          }
+        case _ ⇒ // .. and * are unsupported
```

Comment:
Well, that is the purpose of sending to deadLetters, is it not? So far we have used this approach consistently for all message sends that cannot be completed (wherever possible). Do you propose that we change this to print more targeted warnings in all cases? The other consideration is that messages ending up in deadLetters is unavoidable in some cases and therefore we allow people to switch off all those cases with a single configuration setting.